### ISSUE TITLE
Keep only unique lines is deleting all lines with duplicates

### ISSUE THREAD

**Issue by daniel-barrows** [qcz/vscode-text-power-tools#76]
## Description
The function: "Keep only unique lines" should detect lines which have duplicates, and keep only one copy of the line, which previously had 2 or more copies. However, the function is deleting all lines with duplicates

## Steps to reproduce

1. In any document, enter the following text:

   > a
   > a
   > &nbsp;

2. Select the lines including both lines containing "a". It doesn't matter if leading or trailing newlines are included in the selection.
3. Open the Command Palette
4. Execute the command "Text Power Tools: Keep only unique lines"

### What happens
The entire selection is deleted.
### What I expected to happen
The selection is replaced with a single line containing "a"

### Variations
I have similar results with multiple distinct duplicate lines, (A A B B =>),<br />or with duplicate lines between non-duplicate lines (A B B C => A C).

## Environment
* Text Power Tools v1.44.0
* VSCode Version: 1.83.1 (system setup)
* VSCode Commit: f1b07bd25dfad64b0167beb15359ae573aecd2cc
* VSCode Date: 2023-10-10T23:48:05.904Z
* Electron: 25.8.4
* ElectronBuildId: 24154031
* Chromium: 114.0.5735.289
* Node.js: 18.15.0
* V8: 11.4.183.29-electron.0
* OS: Windows_NT x64 10.0.19045

## Recommendations
You may want to look into using the code from [bibhasdn's Unique Lines extension](https://marketplace.visualstudio.com/items?itemName=bibhasdn.unique-lines). I've been using it for years without problems and it has always worked well, but I had thought Text Power Tools made it redundant.

**Comment by h0jeZvgoxFepBQ2C** [comment]
Yeah this is pretty confusing, but actually what the command says "keep only lines which are unique" and not "remove duplicates"... There should really be another option to filter out the duplicates? I mean VSCode offers something like this, like "Delete Duplicate lines".

<img width="598" alt="Bildschirmfoto 2023-10-27 um 12 53 14" src="https://github.com/qcz/vscode-text-power-tools/assets/10075/a81e09c5-2af3-4bed-b3e9-6f05674ca986">

But yeah, I also think "keep only unqiue lines" should be renamed to something better, coming from sublime this command behaves really different.

**Comment by spachner** [comment]
Yes, same here. Was expecting same as linux tool 'uniq' is doing. Confusing

**Comment by daniel-barrows** [comment]
Per [h0jeZvgoxFepBQ2C](https://github.com/h0jeZvgoxFepBQ2C)'s remarks, let's call this a usability problem rather than a bug. Frankly, I'm reluctant to have Text Power Tools installed because I'm afraid I'm going to use this function by accident. So what should we call this function to prevent user confusion?

How about "**Keep only unique lines (remove any lines that have duplicates)**"?

**Comment by qcz** [comment]
Command renamed to 'Remove any lines that have duplicates' in 1.51.0.
